relay-metrics/src/protocol.rs

```diff
@@ -599,6 +599,31 @@ pub struct Metric {
     pub tags: BTreeMap<String, String>,
 }
 
+const METRIC_NAME_MAX_LENGTH: u8 = 200;
+
+/// An error used when creating a new metric.
+#[derive(Debug)]
+pub struct MetricInitError {
+    name_len: usize,
```

Comment:
Initially, I set this to `u8`. The return type of `.len()` is `usize` and the additional couple of lines of code and effort of integer parsing (?) didn't seem worth it to set to `u8`, considering it shouldn't happen often. Let me know if you disagree or have other suggestions.